src/selectors/entities/channel_categories.ts

```diff
@@ -42,9 +42,13 @@ export function makeGetCategoriesForTeam(): (state: GlobalState, teamId: string)
 export function makeGetChannelsForAllCategories(): (state: GlobalState, teamId: string) => Channel[] {
     return createSelector(
         (state: GlobalState) => state.entities.channels.channels,
+        getMyChannelMemberships,
         (state: GlobalState, teamId: string) => teamId,
-        (allChannels: IDMappedObjects<Channel>, teamId: string) => {
-            return Object.values(allChannels).filter((channel) => channel.team_id === teamId || channel.team_id === '');
+        (allChannels: IDMappedObjects<Channel>, myMembers: RelationOneToOne<Channel, ChannelMembership>, teamId: string) => {
+            return Object.values(allChannels).
+                filter((channel) => channel.delete_at === 0).
```

Comment:
0/5 I would roll the filters into one using ```&&```

Reply:
Does that have a significant performance difference? I chose to leave them separate for readability, but we could certainly combine them at some point